### PR TITLE
Home

### PR DESCRIPTION
**Pull request by Inclet**
#### What does this PR do?
This feature allows a user to see all java developers and can click on one of them to view their profile.
#### Description of Task to be completed?
- Create home component.
- Create Profile component.
- Added tests for the components.
- Added apollo and graphQl.

#### How should this be manually tested?
- First, clone the repo from https://github.com/Inclet/Javadevs--reactNative.git if you haven't.

- Checkout to `home` branch by typing `git checkout home` in your terminal.

- Run `git pull` to pull all changes in the current branch.

- Open `javadevs.xcworkspace` in your Xcode from ios folder where you cloned the repo.
Then start the app by hitting the Run button in Xcode. Automatically, the welcome screen will show up your ios simulator. By clicking the forward under the welcome text on the first screen you will automatically be directed to About screen. From here, you click the blue button labeled `Let's Go` and it will direct you to auth screen. from here you can choose login button, from here, you can log in with any social account and after successful login, you will see a list of all developers.
#### Screenshots (if appropriate)
Screen Shot 2019-08-20 at 12 10 02
Screen Shot 2019-08-20 at 12 10 38

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
 import React from "react";
 import Routes from "./src/Navigations";
+import ApolloClient from 'apollo-boost';
+import { ApolloProvider } from 'react-apollo';
+
+// const networkInterface = createNetworkInterface({
```
Comment (AlvinMugambi) [inline App.js]: Great work. Remember to avoid pushing commented code.

File: src/Home/home.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <Home />
+    </MockedProvider>
+  );
+
+  it("Should match Snapshot", () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('Should show loader initially', () => {
```
Comment (AlvinMugambi) [inline src/Home/home.test.js]: check on this test, it does not test anything. it should expect something.

File: src/Home/index.js [hunk 3]
```diff
@@ -1,16 +1,132 @@
...
+  ScrollView,
+  FlatList
+} from 'react-native';
+import { Query } from "react-apollo";
+import commonStyle from "../Styles/common.style";
+import themeStyle from "../Styles/theme.style";
+import loader from '../Assets/Images/double-ring-loader.gif';
+import Java_developers from '../graphQL_Queries/getAllJavadevelopers';
+import { TouchableOpacity } from 'react-native-gesture-handler';
 class Home extends Component {
```
Comment (AlvinMugambi) [inline src/Home/index.js:17]: Remember to leave a space between the imports and the logic.

**Review by AlvinMugambi — CHANGES_REQUESTED** [review]
Great work. Check on the comments I have left on your PR

File: src/Home/index.js (hunk 3 above)
Comment (Inclet) [reply src/Home/index.js:17]: I've updated this. But I don't know why it still shows it.

Comment (AlvinMugambi) [reply src/Home/index.js:17]: oh. It's okay now. I can see a space

File: .env [hunk 4]
```diff
@@ -7,3 +7,4 @@ MESSAGING_SENDER_ID=414830577349
 APP_ID=1:414830577349=web:eb834c0580bf7c3a
```
Comment (AlvinMugambi) [inline .env]: kindly gitignore this file to avoid sharing senstive information

**Review by AlvinMugambi — CHANGES_REQUESTED** [review]

File: .env (hunk 4 above)
Comment (Inclet) [reply .env]: Ohhh! I forgot to check if I ignore the .env file.
I'm updating that now.

**Review by AlvinMugambi — APPROVED** [review]

**Pull request merged**